src/lib/common/NotificationFormat.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+*
+* Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Ken Zangelin
+*/
+#include <string.h>
+#include <string>
+#include <sstream>
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+#include "common/NotificationFormat.h"
+#include "common/wsStrip.h"
+
+
+
+/* ****************************************************************************
+*
+* notificationFormatToString - 
+*/
+const char* notificationFormatToString(NotificationFormat format, bool noDefault)
```

Comment:
As a general rule, I think that people (thinking to the kind of potential "audience" of this code) would prefer `std::string` as it provides a better abstraction level with a very little overhead penalty (let's be a bit humble and think that the implementers of std::string did a good job, ok? ;). In this case I think that "big savings in simplicity with little penalty in performance" principle (https://github.com/telefonicaid/fiware-orion/blob/develop/doc/manuals/contribution_guidelines.md#general-principles) applies.

A similar opinion appear in one of the cited referenced (I have filtered out some parts that I think there are unpolite):

> I think the better abstraction is worth it. It took many years to get a good string abstraction into C++. I don't believe that Bjarne Stroustroup, so famous for his conservative "only pay for what you use" dictum, would have permitted an obvious performance killer into the language. Higher abstraction is good.

As with the case of the default values, that doesn't mean that `char*` is forbidden, of course. There are places in which it could make fully sense using it, e.g. string-processing functions.